### PR TITLE
Dev/user model

### PR DESCRIPTION
**Pull request by DennisKraaijeveld**
For marketing and UX purposes I have changed the fullname entry to a first and lastname entry. Not sure if this is the correct way of doing it or maybe you'll want to keep the :name entry and use a helper to split the name in a first and last name

### CODE REVIEW

**Comment by tyrant** [comment]
Ah, great, good start. Let's see ... sure, if it's necessary to have your UI split apart a user's name into first/last, let's do it - though now we'll need to make a few decisions about to store these in our database. Depends - can we always assume that any user will only ever have two names, separated by a single space? You'll know more about the realities of Dutch names than me so chances are, you'll know that better than me. If so, then absolutely, a helper function will do the trick nicely; if not, I'll need to mess around with the database and split the User 'name' column into two separate ones. 

Let me know which of the two-names-with-single-space scenarios works better, and I can make the final changes for this PR... though I don't want to step on your toes here! It's awesome you've made this PR at all. If you'd like to charge ahead with either a Rails helper or a database migration yourself, then absolutely be my guest.

**Comment by DennisKraaijeveld** [comment]
Dutch names are pretty simple, although in some cases dutch first names are constructed like this:

Dirk Jan (First name)
Van den Broek (Last name)

But in many cases it is as simple First and Last name. 

I'll only display the first name of the tutors/students inside the App and for marketing purposes it's nicer and more personal to use the first name as well. I think it is easier to concatenate names to a full_name than to figure out how to split the names from a full_name to a first_name. So if you don't have any issues with it (like spending a huge amount of time) I prefer the first_name and last_name. 

I'll push a migrate to this PR so you can check if everything is fine and merge it when ready 👍

**Comment by DennisKraaijeveld** [comment]
Hope this is the correct syntax

File: db/migrate/20210106161247_change_data_for_name.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class ChangeDataForName < ActiveRecord::Migration[6.0]
+  def change
+    change_table :users do |t|
+      t.remove :name
+      t.string :first_name
+      t.string :last_name
```
Comment (tyrant) [inline db/migrate/20210106161247_change_data_for_name.rb:6]: Oops, forgot an `end` here

**Comment by tyrant** [comment]
Not bad! There is one other little bit you or I or any dev would need to do when writing a migration, and that's to transform the existing data into the new table columns before deleting the old column. t's not a big deal right now simply because we've not actually created any prod data yet, though here's the sort of thing I would do. Instead of:
```
class ChangeDataForName < ActiveRecord::Migration[6.0]
  def change
    change_table :users do |t|
      t.remove :name
      t.string :first_name
      t.string :last_name
    end
  end
end
```
We could do:
```
class ChangeDataForName < ActiveRecord::Migration[6.0]
  def up
    change_table :users do |t|
      t.string :first_name
      t.string :last_name
    end
    
    # This is a fiddly bit and could lead to all kinds of hilarious name bugs. As you've pointed out,
    # a name like "Dirk Jan Van den Broek" has first name "Dirk Jan" and last name "Van den Broek".
    # I can't see us writing a nifty little script that could parse a human-rich kludge like that. So 
    # I'd just assumed going with names with the format "first_name[space]last_name". That'll do for now.
    User.find_each do |user|
      user.first_name = user.name.split(' ')[0]
      user.last_name = user.name.split(' ')[1]
      user.save
    end

    remove_column :users, :name
  end

  # Running and rolling back this migration uses custom SQL data-mashing, which Rails wouldn't be able to
  # automatically figure out on its own, so we can't use #change - we'll need to manually write both 
  # #up and #down.
  def down
     add_column :users, :name, :string

    User.find_each do |user|
      user.name = "#{user.first_name} #{user.last_name}"
      user.save
    end

    remove_column :users, :first_name
    remove_column :users, :last_name
  end
end
```
That's the kind of thing I would do in a Prod-ready code base.

**Pull request closed without merging**